src/SparkFun_u-blox_SARA-R5_Arduino_Library.cpp

```diff
@@ -5648,6 +5718,69 @@ SARA_R5_error_t SARA_R5::getFileBlock(const String& filename, char* buffer, size
 
   return SARA_R5_ERROR_SUCCESS;
 }
+SARA_R5_error_t SARA_R5::getAvailableSize(size_t * size)
+{
+
+  // Get the remaining space available on the modem in bytes
+  char * command;
+  char * response;
+
+  command = sara_r5_calloc_char(strlen(SARA_R5_FILE_SYSTEM_LIST_FILES) + 3);
+  if (command==NULL)
+  {
+    return SARA_R5_ERROR_OUT_OF_MEMORY;
+  }
+
+  sprintf(command, "%s=1", SARA_R5_FILE_SYSTEM_LIST_FILES);
+
+  response = sara_r5_calloc_char(minimumResponseAllocation);
+  if (response == nullptr)
+  {
+    free(command);
+    return SARA_R5_ERROR_OUT_OF_MEMORY;
+  }
+  SARA_R5_error_t err = sendCommandWithResponse(command, SARA_R5_RESPONSE_OK_OR_ERROR,response,
+                                                SARA_R5_STANDARD_RESPONSE_TIMEOUT);
+
+  // Process result
+  if (err != SARA_R5_ERROR_SUCCESS)
+  {
+    if (_printDebug == true)
+    {
+      _debugPort->print(F("getAvailableSize: Fail: Error: "));
+      _debugPort->print(err);
+      _debugPort->print(F("  Response: {"));
+      _debugPort->print(response);
+      _debugPort->println(F("}"));
+    }
+    free(command);
+    free(response);
+    return err;
+  }
+  char *responseStart = strstr(response, "+ULSTFILE:");
```

Comment:
David's changes - which I like - move us away from hard-coded strings in the code. Is there an elegant way to append the ":" to SARA_R5_FILE_SYSTEM_LIST_FILES ? I.e. avoid hard-coding here; also avoid 'cheating' by defining SARA_R5_FILE_SYSTEM_LIST_FILES_RESPONSE (which includes the colon). Maybe create a new helper function which will append the colon to whatever command is passed?